services/calculate.go

```diff
@@ -167,152 +98,155 @@ func GetResponseHeaderScore(url string) (totalScore int, XSSReportURL string, ma
+	maxScore += 5
+	if XFrameValue != "" {
+		XFrameValue = strings.TrimSpace(strings.ToLower(XFrameValue))
+		if XFrameValue == XFrameValues[0] || XFrameValue == XFrameValues[1] {
+			totalScore += 5
+		} else if strings.HasPrefix(XFrameValue, XFrameValues[2]) {
+			totalScore += 4
+		}
+	} else {
+		totalScore++
 	}
-	return
+	return totalScore, maxScore
 }
 
 // GetHSTSScore returns the HTTP Strict-Transport-Security Response Header Score of the URL
-func GetHSTSScore(HSTS string) (score int) {
-	if strings.HasPrefix(HSTS, HSTSValues[0]) {
-		score = 4
-		if strings.Contains(HSTS, HSTSValues[1]) || strings.Contains(HSTS, HSTSValues[2]) {
-			score = 5
+func GetHSTSScore(HSTS string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if HSTS != "" {
+		if strings.HasPrefix(HSTS, HSTSValues[0]) {
+			totalScore += 4
+			if strings.Contains(HSTS, HSTSValues[1]) || strings.Contains(HSTS, HSTSValues[2]) {
+				totalScore++
+			}
 		}
+	} else {
+		totalScore += 2
 	}
-	return
+	return totalScore, maxScore
+}
+
+// GetCSPScore returns the score for Content Security Policy Header
+func GetCSPScore(CSP string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if CSP != "" {
+		totalScore += 5
+	} else {
+		totalScore += 3
+	}
+	return totalScore, maxScore
+}
+
+// GetPKPScore returns the score for Public Key Pinning Header
+func GetPKPScore(PKP string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if PKP != "" {
+		totalScore += 5
+	} else {
+		totalScore += 3
+	}
+	return totalScore, maxScore
 }
 
 // GetReferrerPolicyScore returns the HTTP Referrer-Policy Response Header Score of the URL
-func GetReferrerPolicyScore(ReferrerPolicy string) (score int) {
-	ReferrerPolicy = strings.TrimSpace(strings.ToLower(ReferrerPolicy))
-	if strings.Compare(ReferrerPolicy, ReferrerPolicyValues[0]) == 0 {
-		score = 5
-	} else if strings.Compare(ReferrerPolicy, ReferrerPolicyValues[1]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[2]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[3]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[4]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[5]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[6]) == 0 {
-		score = 4
-	} else if strings.Compare(ReferrerPolicy, ReferrerPolicyValues[7]) == 0 {
-		score = 2
+func GetReferrerPolicyScore(ReferrerPolicy string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if ReferrerPolicy != "" {
+		ReferrerPolicy = strings.TrimSpace(strings.ToLower(ReferrerPolicy))
+		if ReferrerPolicy == ReferrerPolicyValues[0] {
+			totalScore += 5
+		} else if ReferrerPolicy == ReferrerPolicyValues[1] || ReferrerPolicy == ReferrerPolicyValues[2] || ReferrerPolicy == ReferrerPolicyValues[3] || ReferrerPolicy == ReferrerPolicyValues[4] || ReferrerPolicy == ReferrerPolicyValues[5] || ReferrerPolicy == ReferrerPolicyValues[6] {
```

Comment:
Could we consider refactoring this as well?

Reply:
@Shriram-Balaji  This is changed